### DISCUSSION TITLE
How should i use state in callbacks

### DISCUSSION THREAD

**Discussion by Slebee** [pmndrs/valtio#479] [Q&A] [answered]
"We shouldn't use snapshots in callbacks, because snapshots can be stale there."

so. how should i use state in callbacks?

**Comment by dai-shi** [comment, marked as answer]
You can just use the proxy state.
```js
const handleClick = () => {
  ++state.count
}
```
I hope I didn't misunderstand your question.

**Reply by Slebee** [reply]
in eslint-plugin-valtio：
```typescript
const state = proxy({ count: 0 })

function App() {
  const snap = useSnapshot(state)
  const handleClick = () => {
    console.log(snap.count) // This is not recommended as it can be stale.
  }
  return (
    <div>
      {snap.count} <button onClick={handleClick}>click</button> 
    </div>
  )
}
```

i want console snap.count，What do you recommend？

**Reply by Slebee** [reply]
ok, i saw https://github.com/pmndrs/valtio/discussions/290#discussioncomment-1743617，but it really confusing.

**Reply by dai-shi** [reply]
> i want console snap.count，What do you recommend？

If you really want to console.log snap.count, destructure in render function.
```jsx
const state = proxy({ count: 0 })

function App() {
  const snap = useSnapshot(state)
  const count = snap.count
  const handleClick = () => {
    console.log(count) // This is fine
  }
  return (
    <div>
      {count} <button onClick={handleClick}>click</button> 
    </div>
  )
}
```

Yes, it is confusing. This is the hardest part of educating valtio.
